### PR TITLE
feat: wire up player access settings to enforce turn permissions

### PR DESCRIPTION
**Pull request by qfulsher**
https://github.com/dayvar14/obr-draw-steel/issues/52

Changes:
- Wires up `Allow players to modify any turn.` and `Allow players to set their turns if player owned.` settings to the turn tracker. 
- Adds title to `Allow players to set their turns if player owned.` noting that this setting only applies if `Owner Only` OBR permission is enabled. 
- If `canModifyAllTurns` is enabled, then players can modify any token turns. Additionally, the `canSetTurnIfPlayerOwned` setting is marked as disabled. (Hope is this'll clue users into how these settings interact.)
- If `canModifyAllTurns` is disabled, then the players cannot modify token turns.
- If `canSetTurnIfPlayerOwned` is enabled, then players can only update their own tokens. If the `Owner Only` OBR permission isn't set, then the player cannot modify any turns. 
- Adds disabled color to light and dark css theme

### CODE REVIEW

File: src/components/SettingsList/SettingsList.tsx [hunk 1]
```diff
@@ -87,13 +87,15 @@ const SettingsList = () => {
               </label>
             </div>
           </div>
-          <div className='settings-item'>
+          <div className={`settings-item ${unsavedSettings.playerAccess.canModifyAllTurns ? 'settings-item-disabled' : ''}`}
```
Comment (daniel-ayvar) [inline src/components/SettingsList/SettingsList.tsx]: I've been using [clsx](https://www.npmjs.com/package/clsx) to manage multiple classnames when they need to be conditional. Perhaps move to use that? Looks a bit cleaner

```suggestion
          <div className={clsx('settings-item', { settings-item-disabled: unsavedSettings.playerAccess.canModifyAllTurns})}
```

File: src/components/SettingsList/SettingsList.tsx [hunk 2]
```diff
@@ -87,13 +87,15 @@ const SettingsList = () => {
...
+            title="Only applies if the player permission 'Owner Only' is enabled.">
             <p>Allow players to set their turns if player owned.</p>
             <div className='settings-item-input'>
-              <label className='switch'>
+              <label className={`switch ${unsavedSettings.playerAccess.canModifyAllTurns ? 'disabled' : ''}`}>
```
Comment (daniel-ayvar) [inline src/components/SettingsList/SettingsList.tsx]: Same things as above
```suggestion
              <label className={clsx('switch', { disabled: unsavedSettings.playerAccess.canModifyAllTurns })}>
```

File: src/styles/common/_input.scss [hunk 3]
```diff
@@ -48,37 +48,54 @@ select {
...
-    position: absolute;
-    top: 0;
-    right: 0;
-    bottom: 0;
-    left: 0;
-    transition: 0.4s;
-    cursor: pointer;
-    background-color: #ccc;
-}
+    input {
```
Comment (daniel-ayvar) [inline src/styles/common/_input.scss:59]: awesome refactor. was still learning scss when doing this. looks alot better

File: src/styles/common/_input.scss [hunk 4]
```diff
@@ -48,37 +48,54 @@ select {
...
 
-.switch input {
-    opacity: 0;
-    width: 0;
-    height: 0;
-}
+    &.disabled {
+        cursor: not-allowed;
+        .slider:before {
+            background-color: map-get($colors, disabled-primary);
```
Comment (daniel-ayvar) [inline src/styles/common/_input.scss]: not a huge fan of the grey for the disabled knob. To me still looks like its interactable as the background color is still the same. Can we instead do an opacity disable style like the following? This maybe possibly be done for the text as well.

Screenshot 2024-08-12 at 10 32 56 PM

Currently do this for when a token doesnt have a turn.

https://github.com/dayvar14/obr-draw-steel/blob/e25293e4ef794020c8f12178fdb38f8673cb47af/src/styles/components/_token-options.scss#L88-L93

File: src/styles/themes/_theme-dark.scss [hunk 5]
```diff
@@ -2,7 +2,8 @@ $colors: (
   primary: #bb99ff,
   contrast: #ffffff,
   primary-text: #ffffff,
-  background-primary: #000000
+  background-primary: #000000,
+  disabled-primary: grey
```
Comment (daniel-ayvar) [inline src/styles/themes/_theme-dark.scss:7]: Considering that this is used for text as well, can we duplicate this for text? So `disabled-text` for text and `disabled-primary` for elements (the input element).

**Review by daniel-ayvar — CHANGES_REQUESTED** [review]

**Comment by daniel-ayvar** [comment]
Thanks for your contribution! Looks great overall. Just a few comments above.

**Comment by qfulsher** [comment]
@dayvar14 Addressed comments, lmk if anything else needs tweaks. Thanks

**Review by daniel-ayvar — APPROVED** [review]

**Comment by daniel-ayvar** [comment]
Looks good. merging.

**Pull request merged**